components/style/matching.rs

```diff
@@ -458,10 +465,60 @@ trait PrivateMatchMethods: TElement {
         &self,
         context: &mut StyleContext<Self>,
         old_values: &mut Option<Arc<ComputedValues>>,
-        new_values: &mut Arc<ComputedValues>,
+        new_resolved_styles: &mut ResolvedElementStyles,
         _restyle_hint: RestyleHint,
         _important_rules_changed: bool,
     ) {
+        if !self.process_animations_for_style(
+            context,
+            old_values,
+            &mut new_resolved_styles.primary.style.0,
+        ) {
+            return;
+        }
+
+        // If we have modified animation or transitions, we recascade style for this node.
+        let mut rule_node = new_resolved_styles.primary.style.0.rules().clone();
```

Comment:
May be worth exposing a `ResolvedElementStyles::primary_style()` or something so that this reads a bit better, but no big deal.